src/components/TopBar/components/Menu/Menu.scss

```diff
@@ -18,6 +18,11 @@ $activator-variables: (
   height: top-bar-height();
   display: flex;
   align-items: center;
+  margin-right: spacing(tight);
+
+  @include breakpoint-before(layout-width(page-with-nav), false) {
+    margin: 0 spacing(extra-tight);
+  }
```

Comment:
Move the margin to the parent component, so the activator can correctly calculate the position of the popover.